perfkitbenchmarker/sample.py

```diff
@@ -33,10 +35,14 @@ def PercentileCalculator(numbers):
   count = len(numbers_sorted)
   total = sum(numbers_sorted)
   result = {}
-  for percentile in PERCENTILES_LIST:
+  for percentile in percentiles:
+    if percentile < 0.0 or percentile > 100.0:
+      raise ValueError('Invalid percentile %s' % percentile)
+
     percentile_string = 'p%s' % str(percentile)
-    result[percentile_string] = numbers_sorted[
-        int(count * float(percentile) / 100)]
+    index = int(count * float(percentile) / 100.0)
+    index = min(index, count - 1)  # Correction to handle 100th percentile.
```

Comment:
but we can probably require the caller to supply an non-empty array.

Or return an empty result if count is zero.